src/DBConnection.cpp

```diff
@@ -153,7 +153,20 @@ int DBConnection::Open(const FilePath fileName)
    mCheckpointStop = false;
    mCheckpointPending = false;
    mCheckpointActive = false;
-   mCheckpointThread = std::thread([this]{ CheckpointThread(); });
+
+   // Open another connection to the DB to prevent blocking the main thread.
+   //
+   // If it fails, then we won't checkpoint until the main thread closes
+   // the associated DB.
+   sqlite3 *db = nullptr;
+   const auto name = sqlite3_db_filename(mDB, nullptr);
+   if (sqlite3_open(name, &db) == SQLITE_OK &&
```

Comment:
We need to capture the rc code here too.  If there is no checkpointing thread, then later on, on closing, the user could be waiting forever for checkpointing to complete.